src/pyggdrasil/tree_inference/_simulate.py

```diff
@@ -44,15 +47,58 @@ def add_noise_to_perfect_matrix(
             - ``MISSING ENTRY`` if ``missing_entry_rate`` is non-zero
     """
     # RNGs for false positives, false negatives, and missing data
-    rng_false_pos, rng_false_neg, rng_miss = random.split(rng, 3)
-
-    # TODO(Pawel, Gordon): Now probably it's the easiest to draw from random.bernoulli
-    #   matrices corresponding to the false positives and false negatives
-    #   and adjust the mutation matrix accordingly
-
-    # Now the matrix can be adjusted according to missing entry simulation
-
-    raise NotImplementedError("This function needs to be implemented.")
+    rng_false_pos, rng_false_neg, rng_miss, rng_homo_pos, rng_homo_neg = random.split(
+        rng, 5
+    )
+    # make matrix to edit and keep unchanged
+    noisy_mat = matrix.copy()
+
+    # Add False Positives - P(D_{ij} = 1 |E_{ij}=0)=alpha
+    # Generate a random matrix of the same shape as the original
+    rand_matrix = random.uniform(key=rng_false_pos, shape=matrix.shape)
+    # Create a mask of elements that satisfy the condition
+    # (original value equals y and random value is less than p)
+    mask = (matrix == 0) & (rand_matrix < false_positive_rate)
```

Comment:
Cool, trick. For the future, you may also consider [jax.random.choice](https://jax.readthedocs.io/en/latest/_autosummary/jax.random.choice.html)

Reply:
Yes, thank you - discovered this too late.